torch/csrc/jit/codegen/cuda/ir_nodes.cpp

```diff
@@ -1597,6 +1597,43 @@ std::vector<IterDomain*> IterDomain::clone(
   return cloned_domains;
 }
 
+IterType inferIterType(IterDomain* i1, IterDomain* i2) {
```

Comment:
@csarofeen I don't have any concern, but please take a look at here just in case.

Reply:
Only nit I have is changing the name to understand it only handles resolution through serial, broadcast, and trivial reduced domains. In other words it shouldn't be used for like gather/stride IDs. Nice cleanup though.

Reply:
I think for gather, it applies the first rule `X + X = X`. Do we want to do it differently?